src/artstationapplication/ArtStationApplication.java

```diff
@@ -24,958 +25,992 @@
+					pad.setBackgroundColor(gui.NONE);
+					gui.colorPickerBackground.setDisable(true);
+				} else {
+					pad.setBackgroundColor(convertColorToInt(gui.colorPickerBackground.getValue()));
+					gui.colorPickerBackground.setDisable(false);
+				}
+			}
+		});
+
+		gui.colorPickerBackground.setOnAction(new EventHandler() {
+			public void handle(Event t) {
+				pad.setBackgroundColor(convertColorToInt(gui.colorPickerBackground.getValue()));
+			}
+		});
+
+		gui.btnUpArrow.setOnAction(event -> {
+			if (pad.listIndex - 1 < 0 || pad.listIndex - 1 == gui.shapes.size()) {
+				canvas.requestFocus();
+				return;
+			}
+			tasks.push(new Change(Transformation.ORD, pad.listIndex, new float[] { pad.listIndex - 1, pad.listIndex }));
+			swapElements(pad.listIndex, pad.listIndex - 1);
+			canvas.requestFocus();
+		});
+
+		gui.btnDownArrow.setOnAction(event -> {
+			if (pad.listIndex + 1 < 0 || pad.listIndex + 1 >= gui.shapes.size()) {
+				canvas.requestFocus();
+				return;
+			}
+			tasks.push(new Change(Transformation.ORD, pad.listIndex, new float[] { pad.listIndex + 1, pad.listIndex }));
+			swapElements(pad.listIndex, pad.listIndex + 1);
+			canvas.requestFocus();
+		});
+
+		gui.btnDelete.setOnAction(event -> {
+			if (gui.shapes.isEmpty())
+				return;
+			Change currentChange = new Change(Transformation.DEL, pad.listIndex, 0);
+			currentChange.createClone(gui.shapes.get(pad.listIndex));
+			tasks.push(currentChange);
+			deleteShape();
+			canvas.requestFocus();
+		});
+
+		gui.btnReset.setOnAction(event -> {
+			if (!gui.shapes.isEmpty()) {
+				tasks.push(
+						new Change(Transformation.RES, pad.listIndex, gui.shapes.get(pad.listIndex).getResetFloats()));
+				gui.shapes.get(pad.listIndex).reset();
+			}
+			canvas.requestFocus();
+		});
+
+		gui.btnCopy.setOnAction(event -> {
+			if (!gui.shapes.isEmpty()) {
+				copyShape();
+				tasks.push(new Change(Transformation.COP, pad.listIndex, 0));
+			}
+			canvas.requestFocus();
+		});
+
+		gui.selectionModel.selectedIndexProperty().addListener(new ChangeListener<Number>() {
+			public void changed(ObservableValue<? extends Number> changed, Number oldVal, Number newVal) {
+				gui.editMode.setSelected(true);
+				pad.editMode();
+				pad.listIndex = (int) newVal;
+			}
+		});
+
+		ContextMenu dropDown = new ContextMenu();
+
+		MenuItem rename = new MenuItem("Rename");
+		rename.setOnAction(new EventHandler<ActionEvent>() { //selecting rename on the drop down
+			@Override
+			public void handle(ActionEvent arg0) {
+				try {
+					TextInputDialog nameDialog = new TextInputDialog(gui.shapes.get(pad.listIndex).getName());
+					nameDialog.setTitle("Rename");
+					nameDialog.setHeaderText("Rename the shape");
+					nameDialog.setContentText("Enter new name");
+
+					Optional<String> result = nameDialog.showAndWait();
+					if (result.isPresent()) {
+						gui.shapes.get(gui.selectionModel.getSelectedIndex()).setName(result.get());
+						gui.shapeViewer.refresh();
```

Comment:
I see you found a way to get the ListViewer to update after the name is changed, good work! The dialog itself looks slick and works exactly as expected. Currently my save functionality does not save the names that you set. It will be easy to add that functionality, but we would need to cleanse the name input to remove any semicolons (;) and commas (,) because those are the delimiters my load functions looks for when loading each shape. If you want you can add the cleanse functionality to this commit, otherwise I will submit it as a separate issue.

Reply:
I'm correctly a student fixing bugs for a class so if you could create a new issue that would be helpful as I submit a URL to each pull and issue for a assignment

Reply:
Yep not a problem, I think that makes the most sense anyway since this was outside of the scope of the original issue.